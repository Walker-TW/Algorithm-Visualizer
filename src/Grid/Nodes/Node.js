import React from 'react';

const Node = () => {
  const getStyle = {
    width: '20px',
    height: '20px',
    border: '1px solid black',
  };

  return <div style={getStyle} />;
};

export default Node;
