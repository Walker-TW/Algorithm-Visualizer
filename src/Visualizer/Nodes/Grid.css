.Column {
  display: inline-block;
}
.Grid {
  margin: 5px 50% 10px 0;
  width: 100%;
  padding: 27px;
}
