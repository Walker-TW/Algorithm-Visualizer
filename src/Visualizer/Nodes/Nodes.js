import React, { useEffect, useState } from "react";
import Node from "../Node/Node";
import PropTypes from "prop-types";

import "./Grid.css";

const Nodes = (props) => {
  const [grid, setGrid] = useState([]);
  const {
    grid: propsGrid,
    fenceToggle,
    gridHasStart,
    gridHasFinish,
    mouseToggle,
    nodeFlag,
    mouseFlag,
    updateNode,
  } = props;

  useEffect(
    (propsGrid) => {
      return (
        () => {
          setGrid(propsGrid);
        },
        console.log("")
      );
    },
    [propsGrid, grid]
  );

  return (
    <div className="Grid">
      {propsGrid.map((row, colIndex) => {
        return (
          <div className="Column" key={colIndex.toString()}>
            {row.map((node, rowIndex) => (
              <Node
                key={colIndex.toString() + " " + rowIndex.toString()}
                id={`node-${node.gridId.colIndex}-${node.gridId.rowIndex}`}
                gridId={node.gridId}
                gridHasStart={gridHasStart}
                gridHasFinish={gridHasFinish}
                fenceToggle={fenceToggle}
                nodeFlag={nodeFlag}
                mouseFlag={mouseFlag}
                updateNode={updateNode}
                mouseToggle={mouseToggle}
              />
            ))}
          </div>
        );
      })}
    </div>
  );
};

Nodes.propTypes = {
  grid: PropTypes.array.isRequired,
  gridHasStart: PropTypes.bool.isRequired,
  gridHasFinish: PropTypes.bool.isRequired,
  nodeFlag: PropTypes.func.isRequired,
  mouseFlag: PropTypes.func.isRequired,
  mouseToggle: PropTypes.bool.isRequired,
  fenceToggle: PropTypes.bool.isRequired,
};

export default Nodes;
