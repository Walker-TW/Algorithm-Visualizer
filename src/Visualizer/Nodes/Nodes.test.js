import React from 'react';
import Nodes from './Nodes';
import { shallowToJson } from 'enzyme-to-json';

import { shallow, mount, render } from 'enzyme';

const defaultProps = {
  grid: [],
  gridHasStart: false,
  gridHasFinish: false,
  nodeFlag: () => {},
  mouseFlag: () => {},
  mouseToggle: false,
  fenceToggle: false,
  resetStartFinish: () => {},
  updateNode: () => {},
};

describe('<Nodes />', () => {
  const nodes = shallow(<Nodes {...defaultProps} />);
  it('renders', () => {
    expect(shallowToJson(nodes)).toMatchSnapshot();
  });
});
