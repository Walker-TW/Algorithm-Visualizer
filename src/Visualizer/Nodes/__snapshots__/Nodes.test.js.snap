// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nodes /> renders 1`] = `
<div
  className="Grid"
/>
`;
